src/components/implementation/no_interface/scheddev/sl.h

```diff
@@ -174,10 +193,163 @@ sl_cs_exit(void)
 	if (!ps_cas(&sl__globals()->lock.u.v, cached.v, 0))    goto retry;
 }
 
+/*
+ * if tid == 0, just block the current thread; otherwise, create a
+ * dependency from this thread on the target tid (i.e. when the
+ * scheduler chooses to run this thread, we will run the dependency
+ * instead (note that "dependency" is transitive).
+ */
+void sl_thd_block(thdid_t tid);
+/*
+ * if abs_timeout == 0, block forever = sl_thd_block()
+ *
+ * @returns: 0 if the thread is woken up by external events before timeout.
+ *	     +ve - number of cycles elapsed from abs_timeout before the thread
+ *		   was woken up by Timeout module.
```

Comment:
I think the documentation here could talk a little about the second argument, noting that it's an absolute timeout. It's obvious from the function signature, but I think it'd be nice to note.